src/data/moves/move-attrs/ability-give-attr.ts

```diff
@@ -33,4 +35,22 @@ export class AbilityGiveAttr extends MoveEffectAttr {
       && !target.getAbility().hasAttrFlag(AbAttrFlag.UNSUPPRESSABLE_ABILITY)
       && user.getAbility().id !== target.getAbility().id;
   }
+
+  /**
+   * If the user has a {@link DETRIMENTAL_ABILITIES | detrimental ability}, or
+   * the target has a {@link HIGH_VALUE_ABILITIES | high-value ability},
+   * grants (+1) with a 50% chance of additional (+1) to effect score.
+   */
+  override getEffectScore(user: EnemyPokemon, target: Pokemon, _move: Move): number {
+    if (HIGH_VALUE_ABILITIES.includes(user.getAbility().id)) {
+      return -5;
+    }
+
+    const userHasDetrimentalAbility = DETRIMENTAL_ABILITIES.includes(user.getAbility().id);
+    const targetHasHighValueAbility = target
+      .getAbilities({ revealedOnly: true })
+      .some((ab) => !ab.passive && HIGH_VALUE_ABILITIES.includes(ab.ability.id));
+
+    return userHasDetrimentalAbility || targetHasHighValueAbility ? 1 + this.getRandomScore(user, 50) : 0;
```

Comment:
These `+1` and `50%` numbers should probably be descriptive constant values

and not a _magic numbers_